src/event_page.js

```diff
@@ -16,7 +16,22 @@ const getAllRegfoxTabs = async () => {
   return chrome.tabs.query(queryOptions);
 };
 
+const injectPrintButtonScript = (tabId, change) => {
+  // Change event only includes the details that actaully changed. We only want
+  // to inject this script
+  if (!change?.url?.match(/manage\.webconnex\.com\/a\/\d+\/reports\/orders\/\d+\/details/)) {
+    return;
+  }
+
+  chrome.scripting.executeScript({

```

Comment:
Why not make this a real content script? https://developer.chrome.com/docs/extensions/mv3/content_scripts/

Reply:
I tried that at first, but due to the single-page-app shenanigans that RegFox is doing I had a really hard time getting a lock on any of the events that would make the buttons show up without just.. constantly running a while loop in the background looking for elements. Because Extension scripts are locked in a different 'world' than the on-page JS we can't really listen in directly, we can only observe the effects those scripts perform.

Well, content scripts can't listen into the tab events. The tab events tell us exactly what we need to know: when we end up on a detail URL. Thus it seemed like the "most" elegant out of the options available.